nexus/reconfigurator/planning/src/blueprint_builder/builder.rs

```diff
@@ -112,13 +111,8 @@ pub enum EnsureMultiple {
     NotNeeded,
 }
 
-fn zpool_id_to_external_name(zpool_id: ZpoolUuid) -> anyhow::Result<ZpoolName> {
-    let pool_name_generated =
-        illumos_utils::zpool::ZpoolName::new_external(zpool_id).to_string();
-    let pool_name = ZpoolName::from_str(&pool_name_generated).map_err(|e| {
-        anyhow!("Failed to create zpool name from {zpool_id}: {e}")
-    })?;
-    Ok(pool_name)
+fn zpool_id_to_external_name(zpool_id: ZpoolUuid) -> ZpoolName {
```

Comment:
Not sure this function is useful anymore - maybe remove and inline `ZpoolName::new_external()` everywhere?

Reply:
Done